File: library/src/main/java/org/meganekkovr/xml/OpacityHandler.java
package org.meganekkovr.xml;

import android.content.Context;
import android.support.annotation.NonNull;

import org.meganekkovr.Entity;

/**
 * Define {@code opacity} attribute.
 */
class OpacityHandler implements XmlAttributeParser.XmlAttributeHandler {

    @NonNull
    @Override
    public String attributeName() {
        return "opacity";
    }

    @Override
    public void parse(@NonNull Entity entity, @NonNull String rawValue, @NonNull Context context) {

        float opacity = Float.parseFloat(rawValue);
        entity.setOpacity(opacity);
    }
}

File: library/src/main/java/org/meganekkovr/xml/OpacityHandler.kt
package org.meganekkovr.xml

import android.content.Context

import org.meganekkovr.Entity

/**
 * Define `opacity` attribute.
 */
internal class OpacityHandler : XmlAttributeParser.XmlAttributeHandler {

    override fun attributeName(): String {
        return "opacity"
    }

    override fun parse(entity: Entity, rawValue: String, context: Context) {

        val opacity = rawValue.toFloatOrNull() ?: return
        entity.opacity = opacity
    }
}
